### PR TITLE
Add config layer basic component

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
* Add "config layer" basic component which can be triggered from "layer control" panel or "add new layer" modal
* Add show/hide layer control component
* Add Zoom in/out component on maplibre control

### Screenshot
image

image

image

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/plugins/maps_dashboards/common/index.ts [hunk 1]
```diff
@@ -20,3 +20,10 @@ export const MAP_INITIAL_STATE = {
 export const APP_PATH = {
   CREATE_MAP: '/create-map',
 };
+
+export enum LAYER_TYPE {
+  BASE_MAP = 'base map',
+  CLUSTER_MAP = 'cluster',
+  CHOROPLETH_MAP = 'choropleth',
+  WEB_MAP_SERVICE = 'wms',
```
Comment (VijayanB) [inline src/plugins/maps_dashboards/common/index.ts:28]: Let's add this when we add new layer

File: src/plugins/maps_dashboards/public/components/layer_control_panel/layer_control_panel.tsx [hunk 2]
```diff
@@ -12,79 +12,150 @@ import {
...
+
   // TODO: replace it once layers model ready
-  const demoLayers = [
+  const demoLayers: ILayerConfig[] = [
     {
       label: 'Base Map',
       iconType: 'visMapRegion',
+      id: 'example_id_1',
+      type: LAYER_TYPE.BASE_MAP,
+      name: 'Base Map Layer',
```
Comment (VijayanB) [inline src/plugins/maps_dashboards/public/components/layer_control_panel/layer_control_panel.tsx:42]: Shall we remove Layer suffix?

**Review by VijayanB — APPROVED** [review]
LGTM. Added some minor comment.

File: src/plugins/maps_dashboards/public/components/layer_control_panel/layer_control_panel.tsx (hunk 2 above)
Comment (junqiu-lei) [reply src/plugins/maps_dashboards/public/components/layer_control_panel/layer_control_panel.tsx:42]: Yes, that's a example, will update when integrated with layer models

File: src/plugins/maps_dashboards/public/model/ILayerConfig.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+//TODO: this is used for UI component development, remove or update it once layer model ready
```
Comment (martin-gaievski) [inline src/plugins/maps_dashboards/public/model/ILayerConfig.ts:2]: do we have a github issue or sprint task to address this todo item?

Comment (junqiu-lei) [reply src/plugins/maps_dashboards/public/model/ILayerConfig.ts:2]: Yes, just created a gihub issue to track this. https://github.com/opensearch-project/dashboards-maps/issues/56 Thanks!

**Pull request merged**